infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -136,19 +139,37 @@ private void clearTables() {
             log.debug("Clear shadow streams, count={}", streamViewMap.size());
         }
 
-        try (TxnContext txnContext = logReplicationMetadataManager.getTxnContext()) {
-            logReplicationMetadataManager.appendUpdate(txnContext, LogReplicationMetadataType.TOPOLOGY_CONFIG_ID, topologyConfigId);
+        clearStreams();
+    }
 
-            for (UUID streamID : streamViewMap.keySet()) {
-                UUID streamToClear = streamID;
-                if (phase == Phase.TRANSFER_PHASE) {
-                    streamToClear = regularToShadowStreamId.get(streamID);
+    private void clearStreams() {
+        try {
+            IRetry.build(IntervalRetry.class, () -> {
+                try (TxnContext txnContext = logReplicationMetadataManager.getTxnContext()) {
+                    logReplicationMetadataManager.appendUpdate(txnContext, LogReplicationMetadataType.TOPOLOGY_CONFIG_ID, topologyConfigId);
+                    clearStreams(txnContext);
+                    txnContext.commit();
+                } catch (TransactionAbortedException tae) {
+                    log.error("Error while attempting to clear tables.", tae);
+                    throw new RetryNeededException();
                 }
+                return null;
```

Comment:
this return null means this part of code will only run one time, if there's no return, this part of code will retry many times?

Reply:
The retryFunction expects a return value, since we don't have any return value we can just return null. The part of the code that will know a retry is required is line 154: throw new RetryNeededException(), which we throw if we catch a TransactionAbortedException, so the transaction is retried.